elrond-config.md

```diff
@@ -538,37 +551,160 @@ TODO: Implement [reserved keys and read-only runtimes](https://github.com/Elrond
+      requires ACCTFROM =/=K ACCTTO andBool VALUE <=Int ORIGFROM
+      [priority(60)]
+
+    rule <commands> transferESDT(ACCTFROM, ACCTTO, esdtTransfer(TOKEN, VALUE, 0)) 
+                 => . ... 
+         </commands>
+         <account>
+           <address> ACCTFROM </address>
+           <esdtData>
+             <esdtId> TOKEN </esdtId>
+             <esdtBalance> ORIGFROM => ORIGFROM -Int VALUE </esdtBalance>
+             <frozen> false </frozen>
+           </esdtData>
+           ...
+         </account>
+         <account>
+           <address> ACCTTO </address>
+           (.Bag => <esdtData>
+             <esdtId> TOKEN </esdtId>
+             <esdtBalance> VALUE </esdtBalance>
+             <frozen> false </frozen>
+           </esdtData>)
+           ...
+         </account>
+      requires ACCTFROM =/=K ACCTTO andBool VALUE <=Int ORIGFROM
+      [priority(61)]
+
     rule <commands> #transferSuccess => . ... </commands>
          <instrs> . </instrs>
 ```
 
 ## Calling Contract
 
 ```k
-    syntax InternalCmd ::= callContract ( Bytes, Bytes, Int,     String, List, Int, Int ) [klabel(callContractString)]
-                         | callContract ( Bytes, Bytes, Int, WasmString, List, Int, Int ) [klabel(callContractWasmString)]
-                         | mkCall       ( Bytes, Bytes, Int, WasmString, List, Int, Int )
+    syntax InternalCmd ::= callContract ( Bytes, Bytes, Int, List,     String, List, Int, Int ) [klabel(callContractString)]
+                         | callContract ( Bytes, Bytes, Int, List, WasmString, List, Int, Int ) [klabel(callContractWasmString)]
+                         | mkCall       ( Bytes, Bytes, Int, List, WasmString, List, Int, Int )
  // -------------------------------------------------------------------------------------
-    rule <commands> callContract(FROM, TO, VALUE, FUNCNAME:String, ARGS, GASLIMIT, GASPRICE)
-                 => callContract(FROM, TO, VALUE, #unparseWasmString("\"" +String FUNCNAME +String "\""), ARGS, GASLIMIT, GASPRICE)
+    rule <commands> callContract(FROM, TO, VALUE, ESDT, FUNCNAME:String, ARGS, GASLIMIT, GASPRICE)
+                 => callContract(FROM, TO, VALUE, ESDT, #unparseWasmString("\"" +String FUNCNAME +String "\""), ARGS, GASLIMIT, GASPRICE)
                     ...
          </commands>
       [priority(60)]
 
-    rule <commands> callContract(FROM, TO, VALUE, FUNCNAME:WasmStringToken, ARGS, GASLIMIT, GASPRICE)
+    // The memory instance linked to the contract's account should be restored to the initial state after the contract call.
+    rule <commands> callContract(FROM, TO, VALUE, ESDT, FUNCNAME:WasmStringToken, ARGS, GASLIMIT, GASPRICE)
                  => pushWorldState
                  ~> transferFunds(FROM, TO, VALUE)
-                 ~> mkCall(FROM, TO, VALUE, FUNCNAME, ARGS, GASLIMIT, GASPRICE)
+                 ~> transferESDTs(FROM, TO, ESDT)
+                 ~> mkCall(FROM, TO, VALUE, ESDT, FUNCNAME, ARGS, GASLIMIT, GASPRICE)
```

Comment:
Does the memory need to be emptied when make a call?

Reply:
@yzhang90 
When a module is initialized (during SC deployment), the memory consists of constants like static strings and empty sections. This rule assumes the memory is in this initial state, so it does not empty it _before_ the `mkCall`.

We only clean up the memory after `#endWasm` is reached using the `#restoreMem(_)` helper.

Reply:
> How do calls between contracts work? I'm assuming that the called contract needs its own "fresh" memory.

Added a TODO comment about this. Each contract call should run its own wasm module instance, and there should be a `<callStack>` to keep intermediate `<callState>`'s.